docs/src/snap/howto/2-node-ha.md

```diff
@@ -0,0 +1,417 @@
+can achieve high availability with just two nodes while using the default
+datastore, Dqlite.
+
+Dqlite cannot achieve Raft quorum with less than three nodes. This means that
+Dqlite will not be able to replicate data and the secondaries will simply
+forward the queries to the primary node.
+
+In the event of a node failure, database recovery will require following the
+steps in the [Dqlite recovery guide].
+
+## Proposed solution
+
+Since Dqlite data replication is not available in this situation, we propose
+using synchronous block level replication through
+[Distributed Replicated Block Device] (DRBD).
+
+The cluster monitoring and failover process will be handled by Pacemaker and
+Corosync. After a node failure, the DRBD volume will be mounted on the standby
+node, allowing access to the latest Dqlite database version.
+
+Additional recovery steps are automated and invoked through Pacemaker.
+
+## Alternatives
+
+Another possible approach is to use PostgreSQL with Kine and logical
+replication. However, it is outside the scope of this document.
+
+See the [external datastore guide] for more information on how Canonical
+Kubernetes can be configured to use other datastores.
+
+## Guide
+
+### Prerequisites
+
+* Ensure both nodes are part of the Kubernetes cluster.
+  See the [getting started] and [add/remove nodes] guides.
+* The user associated with the HA service has SSH access to the peer node and
+  passwordless sudo configured. For simplicity, the default "ubuntu" user can
+  be used.
+* We recommend using static IP configuration.
+
+The [2ha.sh script] automates most operations related to the 2-node HA scenario
+and is included in the snap.
+
+The first step is to install the required packages:
+
+```
+/snap/k8s/current/k8s/hack/2ha.sh install_packages
+```
+
+### DRBD
+
+For the purpose of this guide, we are going to use a loopback device as DRBD
+backing storage:
+
+```
+sudo dd if=/dev/zero of=/opt/drbd0-backstore bs=1M count=2000
+```
+
+Ensure that the loopback device is attached at boot time, before Pacemaker
+starts.
+
+```
+cat <<EOF | sudo tee /etc/rc.local
+#!/bin/sh
+mknod /dev/lodrbd b 7 200
+losetup /dev/lodrbd /opt/drbd0-backstore
+EOF
+
+sudo chmod +x /etc/rc.local
+
+cat <<EOF | sudo tee /etc/systemd/system/rc-local.service
+# This unit gets pulled automatically into multi-user.target by
+# systemd-rc-local-generator if /etc/rc.local is executable.
+[Unit]
+Description=/etc/rc.local Compatibility
+Documentation=man:systemd-rc-local-generator(8)
+ConditionFileIsExecutable=/etc/rc.local
+After=network.target
+
+[Service]
+Type=forking
+ExecStart=/etc/rc.local start
+TimeoutSec=0
+RemainAfterExit=yes
+GuessMainPID=no
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+sudo systemctl enable rc-local.service
+sudo systemctl start rc-local.service
+```
+
+Let's configure the DRBD block device that will hold the Dqlite data.
+Ensure the correct node addresses are used.
```

Comment:
```suggestion
Ensure the correct node addresses are used.
```
This is important, instead of repeating it later make it stand out by putting it in a note box. Example for a note: https://github.com/canonical/k8s-snap/blob/c74b9f580f92b2fc6b77bf4ad692eb348b62a17f/docs/src/charm/howto/charm.md?plain=1#L18-L21

Reply:
Those are just some reminders to fill in the addresses instead of copy/pasting the following by mistake:

```
HAONE_ADDR=<firstNodeAddress>
HATWO_ADDR=<secondNodeAddress>
```

Do we need a note box for that?